chartpress.py

```diff
@@ -292,6 +294,38 @@ def _strip_identifiers_build_suffix(identifier):
     return re.sub(r'[-\.]\d{3,}\.\w{4,}\Z', "", identifier)
 
 
+def _get_published_chart_information(chart_repo_url, chart_name, chart_version):
+    """
+    Provide information about a chart release from a chart repository url along
+    with a chart name and version to search for.
+
+    - Returns the dictionary representing the matching chart entry from the Helm
+      chart repository if a matching entry is found.
+    - Returns None if a Helm chart repostiory is found but the matching chart
+      name and version isn't.
+    - Raises errors if the Helm chart repository isn't accessible or doesn't
+      seem to be one.
+    """
+    index_url = urljoin(chart_repo_url + "/", "index.yaml")
+    response = requests.get(index_url)
+
+    # this can throw errors, and doing that makes sense
+    # - connection failure
+    # - response doesn't seem like a Helm chart repository
+    response.raise_for_status()
+    index = yaml.load(response.content)
+    chart_repository_entries = index["entries"]
+
+    # let's not throw errors now, but instead return None if
+    # we fail to find a matching chart version
+    chart_releases = chart_repository_entries.get(chart_name, None)
+    if not chart_releases:
+        return None
+
+    matching_chart_release = dict(next((r for r in chart_releases if r["version"]==chart_version), {}))
```

Comment:
What's this `dict` call doing?

Perhaps clearer:

```suggestion
    matching_chart_release = None
    for release in chart_releases:
         if release["version"] == chart_version:
            matching_chart_release = release
            break
```